Cool, thanks @helrond - incorporated those changes and another change to get the styling together. The build is failing in a way I don't understand: 
```FAIL: test_bagitprofiles (bag_transfer.test.test_bagitprofiles.BagItProfileTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/code/aurora/bag_transfer/test/test_bagitprofiles.py", line 195, in test_bagitprofiles
    self.assertEqual(update_request.status_code, 302, "Request was not redirected")
AssertionError: 200 != 302 : Request was not redirected

--- a/aurora/bag_transfer/orgs/views.py
+++ b/aurora/bag_transfer/orgs/views.py
@@ -137,30 +137,11 @@ class BagItProfileManageView(TemplateView):
 
     def post(self, request, *args, **kwargs):
         instance = None
-        organization = Organization.objects.get(pk=self.kwargs.get("pk"))
         if self.kwargs.get("profile_pk"):
             instance = get_object_or_404(BagItProfile, pk=self.kwargs.get("profile_pk"))
         form = BagItProfileForm(request.POST, instance=instance)
         if form.is_valid():
-            if BagItProfile.objects.filter(
-                applies_to_organization=form.cleaned_data["applies_to_organization"],
-                contact_email=form.cleaned_data["contact_email"],
-                source_organization=form.cleaned_data["source_organization"],
-                version=form.cleaned_data["version"],
-                bagit_profile_identifier=form.cleaned_data["bagit_profile_identifier"],
-                external_description=form.cleaned_data["external_description"],
-                serialization=form.cleaned_data["serialization"],
-            ).exists():
-                bagit_profile = BagItProfile.objects.filter(
-                    applies_to_organization=form.cleaned_data["applies_to_organization"],
-                    contact_email=form.cleaned_data["contact_email"],
-                    source_organization=form.cleaned_data["source_organization"],
-                    version=form.cleaned_data["version"],
-                    bagit_profile_identifier=form.cleaned_data["bagit_profile_identifier"],
-                    external_description=form.cleaned_data["external_description"],
-                    serialization=form.cleaned_data["serialization"])[0]
-            else:
-                bagit_profile = form.save()
+            bagit_profile = instance if instance else form.save()
             bag_info_formset = BagItProfileBagInfoFormset(
                 request.POST, instance=bagit_profile, prefix="bag_info")
             manifests_allowed_formset = ManifestsAllowedFormset(
@@ -186,7 +167,6 @@ class BagItProfileManageView(TemplateView):
             ]
             for formset in forms_to_save:
                 if not formset.is_valid():
-                    print(formset.non_form_errors())
                     messages.error(
                         request,
                         "There was a problem with your submission. Please correct the error(s) below and try again.",
@@ -219,12 +199,9 @@ class BagItProfileManageView(TemplateView):
             bagit_profile.save()
             messages.success(
                 request,
-                "BagIt Profile for {} saved".format(
-                    bagit_profile.applies_to_organization.name
-                ),
+                "BagIt Profile for {} saved".format(bagit_profile.applies_to_organization.name),
             )
             return redirect("orgs:detail", bagit_profile.applies_to_organization.pk)
-        print(form.errors)
         messages.error(
             request,
             "There was a problem with your submission. Please correct the error(s) below and try again.",
@@ -234,7 +211,7 @@ class BagItProfileManageView(TemplateView):
             self.template_name,
             {
                 "form": BagItProfileForm(request.POST, instance=instance),
-                "organization": organization,
+                "organization": Organization.objects.get(pk=self.kwargs.get("pk")),
                 "bag_info_formset": BagItProfileBagInfoFormset(
                     request.POST, prefix="bag_info"),
                 "manifests_allowed_formset": ManifestsAllowedFormset(

--- a/aurora/bag_transfer/orgs/views.py
+++ b/aurora/bag_transfer/orgs/views.py
@@ -106,24 +106,18 @@ class BagItProfileManageView(TemplateView):
                 }
             )
         bag_info_formset = BagItProfileBagInfoFormset(
-            instance=profile, prefix="bag_info"
-        )
+            instance=profile, prefix="bag_info")
         manifests_allowed_formset = ManifestsAllowedFormset(
-            instance=profile, prefix="manifests_allowed"
-        )
+            instance=profile, prefix="manifests_allowed")
         manifests_formset = ManifestsRequiredFormset(
-            instance=profile, prefix="manifests"
-        )
+            instance=profile, prefix="manifests")
         serialization_formset = AcceptSerializationFormset(
-            instance=profile, prefix="serialization"
-        )
+            instance=profile, prefix="serialization")
         version_formset = AcceptBagItVersionFormset(instance=profile, prefix="version")
         tag_manifests_formset = TagManifestsRequiredFormset(
-            instance=profile, prefix="tag_manifests"
-        )
+            instance=profile, prefix="tag_manifests")
         tag_files_formset = TagFilesRequiredFormset(
-            instance=profile, prefix="tag_files"
-        )
+            instance=profile, prefix="tag_files")
         return render(
             request,
             self.template_name,
@@ -158,41 +152,29 @@ class BagItProfileManageView(TemplateView):
                 serialization=form.cleaned_data["serialization"],
             ).exists():
                 bagit_profile = BagItProfile.objects.filter(
-                    applies_to_organization=form.cleaned_data[
-                        "applies_to_organization"
-                    ],
+                    applies_to_organization=form.cleaned_data["applies_to_organization"],
                     contact_email=form.cleaned_data["contact_email"],
                     source_organization=form.cleaned_data["source_organization"],
                     version=form.cleaned_data["version"],
-                    bagit_profile_identifier=form.cleaned_data[
-                        "bagit_profile_identifier"
-                    ],
+                    bagit_profile_identifier=form.cleaned_data["bagit_profile_identifier"],
                     external_description=form.cleaned_data["external_description"],
-                    serialization=form.cleaned_data["serialization"],
-                )[0]
+                    serialization=form.cleaned_data["serialization"])[0]
             else:
                 bagit_profile = form.save()
             bag_info_formset = BagItProfileBagInfoFormset(
-                request.POST, instance=bagit_profile, prefix="bag_info"
-            )
+                request.POST, instance=bagit_profile, prefix="bag_info")
             manifests_allowed_formset = ManifestsAllowedFormset(
-                request.POST, instance=bagit_profile, prefix="manifests_allowed"
-            )
+                request.POST, instance=bagit_profile, prefix="manifests_allowed")
             manifests_formset = ManifestsRequiredFormset(
-                request.POST, instance=bagit_profile, prefix="manifests"
-            )
+                request.POST, instance=bagit_profile, prefix="manifests")
             serialization_formset = AcceptSerializationFormset(
-                request.POST, instance=bagit_profile, prefix="serialization"
-            )
+                request.POST, instance=bagit_profile, prefix="serialization")
             version_formset = AcceptBagItVersionFormset(
-                request.POST, instance=bagit_profile, prefix="version"
-            )
+                request.POST, instance=bagit_profile, prefix="version")
             tag_manifests_formset = TagManifestsRequiredFormset(
-                request.POST, instance=bagit_profile, prefix="tag_manifests"
-            )
+                request.POST, instance=bagit_profile, prefix="tag_manifests")
             tag_files_formset = TagFilesRequiredFormset(
-                request.POST, instance=bagit_profile, prefix="tag_files"
-            )
+                request.POST, instance=bagit_profile, prefix="tag_files")
             forms_to_save = [
                 bag_info_formset,
                 manifests_allowed_formset,
@@ -204,6 +186,7 @@ class BagItProfileManageView(TemplateView):
             ]
             for formset in forms_to_save:
                 if not formset.is_valid():
+                    print(formset.non_form_errors())
                     messages.error(
                         request,
                         "There was a problem with your submission. Please correct the error(s) below and try again.",
@@ -253,26 +236,19 @@ class BagItProfileManageView(TemplateView):
                 "form": BagItProfileForm(request.POST, instance=instance),
                 "organization": organization,
                 "bag_info_formset": BagItProfileBagInfoFormset(
-                    request.POST, prefix="bag_info"
-                ),
+                    request.POST, prefix="bag_info"),
                 "manifests_allowed_formset": ManifestsAllowedFormset(
-                    request.POST, prefix="manifests_allowed"
-                ),
+                    request.POST, prefix="manifests_allowed"),
                 "manifests_formset": ManifestsRequiredFormset(
-                    request.POST, prefix="manifests"
-                ),
+                    request.POST, prefix="manifests"),
                 "serialization_formset": AcceptSerializationFormset(
-                    request.POST, prefix="serialization"
-                ),
+                    request.POST, prefix="serialization"),
                 "version_formset": AcceptBagItVersionFormset(
-                    request.POST, prefix="version"
-                ),
+                    request.POST, prefix="version"),
                 "tag_manifests_formset": TagManifestsRequiredFormset(
-                    request.POST, prefix="tag_manifests"
-                ),
+                    request.POST, prefix="tag_manifests"),
                 "tag_files_formset": TagFilesRequiredFormset(
-                    request.POST, prefix="tag_files"
-                ),
+                    request.POST, prefix="tag_files"),
                 "meta_page_title": "BagIt Profile",
             },
         )

--- a/aurora/bag_transfer/test/test_bagitprofiles.py
+++ b/aurora/bag_transfer/test/test_bagitprofiles.py
@@ -186,10 +186,9 @@ class BagItProfileTestCase(TestCase):
                 "manifests-INITIAL_FORMS": 1,
                 "manifests-0-id": 1,
                 "manifests-0-DELETE": True,
-                "manifests_allowed-TOTAL_FORMS": 0,
-                "manifests_allowed-INITIAL_FORMS": 1,
-                "manifests_allowed-0-id": 1,
-                "manifests_allowed-0-DELETE": True,
+                "manifests_allowed-TOTAL_FORMS": 1,
+                "manifests_allowed-INITIAL_FORMS": 0,
+                "manifests_allowed-0-name": random.choice(["sha256", "sha512"]),
             },
         )
         self.assertEqual(update_request.status_code, 302, "Request was not redirected")

